gbn/timeout_manager.go

```diff
@@ -435,6 +493,8 @@ func (m *TimeoutManager) SetStaticResendTimeout(resendTimeout time.Duration) {
 	defer m.timeoutManagerMu.Unlock()
 
 	m.resendTimeout = resendTimeout
+	m.resendBooster.SetOriginalTimeout(resendTimeout)
```

Comment:
instead set this timeout at the time of initialising the resendBooster. You can do this by just re-shuffling the  order of things in `NewTimeOutManager`: 

1: construct TimeoutMgr
2. Apply it's functional options.
3. initialise boosters

Reply:
Thanks! Good idea, I've implemented this with the latest push. It required me to now add the `resendBoostPercent` & `handshakeTimeout` fields to the `TimeoutManager` though, which isn't ideal. But I think the new version is much better than the last version though, so I definitely think that's worth it.